org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/points/ScanView.java

```diff
@@ -771,4 +763,36 @@ public void itemRemoved(SeriesItemEvent evt) {
 		
 	}
 
+	private List<IScanPathModel> getPath() {
+		return pointsFilter.getModels(seriesTable.getSeriesItems());
+	}
+
+	private List<IScanPathModel> setPath(List<IScanPathModel> path) {
+		List<IScanPathModel> old = pointsFilter.getModels(seriesTable.getSeriesItems());
+		try {
+			this.saved = pointsFilter.createDescriptors(path);
+		} catch (GeneratorException e) {
+			logger.error("Unable to create descriptors for "+path);
+		}
+		return old;
+	}
+
+	@Override
+	public <T> boolean isSeriesOf(Class<T> class1) {
+		return class1==IPointGenerator.class;
+	}
+
+	@Override
+	public ISeriesItemDescriptor find(Predicate<ISeriesItemDescriptor> predicate) {
+		return seriesTable.find(predicate);
+	}
+
+	public static String getLastPath() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Synchronize this method to match the synchronization on "setLastPath". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2886)